core/schema/env.graphqls

```diff
@@ -0,0 +1,98 @@
+extend type Query {
+  "Load a environment from ID."
+  environment(id: EnvironmentID): Environment!
+
+  "Load a environment check from ID."
+  environmentCheck(id: CheckID): Check!
+
+  "Return the current environment being executed in."
+  currentEnvironment: Environment!
+}
+
+"A unique environment identifier."
+scalar EnvironmentID
+
+"A unique environment check identifier."
+scalar CheckID
+
+"A unique environment check result identifier."
+scalar CheckResultID
+
+"A group of Dagger entrypoints that can be queried and/or invoked."
+type Environment {
```

Comment:
Our longer term idea is for this to become an `interface` instead. I didn't include that here yet because I want to reserve the right to save that for follow up PRs.

The reason being that I *suspect* adding support for graphql interfaces may end up being a very significant chunk of work since it requires updates to the codegen for every SDK. That work will end up being worth it (I imagine we will have plenty of other use cases besides this one), but I don't necessarily want this PR to get blocked on that effort.

Reply:
This would be pretty tricky to implement though but I can def see the benefit of supporting interface

Reply:
I think the rough outline would be that when there's a gql `interface`, the codegen produces whatever is closest to that concept in it's language:
1. For Go and Typescript, it's literally just an `interface` in those language's syntax (ts has those iirc, right?)
2. For Python, I guess it might literally just be a type annotation since the underlying objects are all dynamic anyways. Not sure though
3. For Rust, a trait? Again, not sure
4. etc.

I think that actually may be all that's fundamentally needed.

It would at least be nice-to-have the ability to also convert the interface into a concrete object (possible in [raw graphql](https://www.apollographql.com/docs/apollo-server/schema/unions-interfaces/#querying-an-interface), in go it would be `foo, ok := ifaceObject.(Foo)`, etc.). But I'm not sure if there is actually a hard requirement for that, so hopefully could be left out for sake of simplicity.

So honestly it _might_ not be too much work for an individual SDK, at least in this early "theoretical" stage. I'm just more concerned that it's `work * numSDKs` :-)

Reply:
> For Go and Typescript, it's literally just an interface in those language's syntax (ts has those iirc, right?)

For typescript it would be an `abstract class` or an interface, depending on what we want at the end.

> For Rust, a trait? Again, not sure

Yep, a `trait` shall do the job

> So honestly it might not be too much work for an individual SDK, at least in this early "theoretical" stage. I'm just more concerned that it's work * numSDKs :-)

Yeah totally but once it's done, we shall not touch it anymore, I think it's worth the time spent, I'm just scared that it change API at some point, I might be wrong tho. 

I do not have a clear view of how API shall look using interface